File: frontend/src/types/graph.ts
export type GraphNode = {
  entity_id: string;
  display_name: string;
  entity_type: string;
  status: string;
  sensitivity: string;
  is_focal: boolean;
};

export type GraphEdge = {
  edge_id: string;
  from_entity_id: string;
  to_entity_id: string;
  relation_type: string;
  directed: boolean;
  status: string;
  confidence: number;
  sensitivity: string;
};

export type EgoGraph = {
  focal_entity_id: string;
  depth: number;
  nodes: GraphNode[];
  edges: GraphEdge[];
  filters_applied: Record<string, string | number>;
};

export type GraphFilters = {
  relation_type: string;
  status: string;
  sensitivity: string;
};

File: backend/src/kinlayer_backend/schemas/graph.py
from kinlayer_backend.schemas.common import APIModel


class GraphNode(APIModel):
    entity_id: str
    display_name: str
    entity_type: str
    status: str
    sensitivity: str
    is_focal: bool = False


class GraphEdge(APIModel):
    edge_id: str
    from_entity_id: str
    to_entity_id: str
    relation_type: str
    directed: bool
    status: str
    confidence: float
    sensitivity: str


class EgoGraph(APIModel):
    focal_entity_id: str
    depth: int
    nodes: list[GraphNode]
    edges: list[GraphEdge]
    filters_applied: dict[str, str | int]
